main.go

```diff
@@ -0,0 +1,97 @@
+package main
+
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"sort"
+	"strconv"
+	"strings"
+)
+
+func main() {
+	fieldsParameter := flag.String("f", "", "Choose field to print")
+	flag.Parse()
+
+	content, readAllErr := ioutil.ReadAll(os.Stdin)
+	if readAllErr != nil {
+		fmt.Fprintln(os.Stderr, readAllErr)
+		os.Exit(1)
+	}
+
+	if len(content) <= 0 {
+		return
+	}
+
+	input := strings.Split(*fieldsParameter, ",")
+	inputAsInt := make([][]int, len(input))
+
+	for i := range input {
+		split := strings.Split(input[i], "-")
+		inputAsInt[i] = make([]int, len(split))
+
+		for sid, s := range split {
+			value, err := strconv.Atoi(s)
+			if err != nil {
+				fmt.Fprintln(os.Stderr, err)
+				os.Exit(1)
+			}
+			inputAsInt[i][sid] = value
+		}
+
+		if inputAsInt[i][1]-inputAsInt[i][0] > 1 {
+			for p := inputAsInt[i][0] + 1; p < inputAsInt[i][1]; p++ {
+				inputAsInt[i] = append(inputAsInt[i], p)
+			}
+			sort.Ints(inputAsInt[i])
+		}
+
+	}
+
+	var data, parameterBasedData [][][]byte
+	var c, counter int
+	delimiter := []byte{9}
+
+	// Fieldparameter
+	contentLineSplit := bytes.Split(content, []byte{'\n'})
+	data = make([][][]byte, len(contentLineSplit))
+
+	for i := 0; i < len(contentLineSplit); i++ {
+		data[i] = bytes.Split(contentLineSplit[i], delimiter)
+	}
+
+	counter = 0
+	for a := range data {
+		if len(data[a]) > counter {
+			counter = len(data[a])
+		}
+	}
+
+	c = len(data) - 1
+
+	if len(data[c][0]) > 0 {
+		c = len(data)
+	}
+
+	parameterBasedData = data
+
+	for i := 0; i < c; i++ {
+		var output []string
+		for j := range input {
+			for _, k := range inputAsInt[j] {
+				if k > counter {
+					k = counter
```

Comment:
You should just check if k is larger then len(parameterBasedData[i]), because now you cause the following:

test.txt:
```
1	2	3	4	5
11	22	33	44	55
```

Result:
```
go run . -f 1-10 < test.txt
1	2	3	4	5	5	5	5	5	5
11	22	33	44	55	55	55	55	55	55
```

Should be:
```
cut -f 1-10 test.txt
1	2	3	4	5
11	22	33	44	55
```

I don't think you need `counter`